com.unity.netcode.gameobjects/Components/NetworkTransform.cs

```diff
@@ -844,25 +944,61 @@ private void Awake()
             }
         }
 
+        /// <summary>
+        /// These local values are for the non-authoritative side
+        /// This prevents non-authoritative instances from changing
+        /// the transform.
+        /// </summary>
+        private Vector3 m_LocalPosition;
+        private Vector3 m_LocalScale;
+        private Quaternion m_LocalRotation;
+
+        /// <summary>
+        /// Applies the last authorized position, scale, and rotation
+        /// </summary>
+        private void ApplyTransformValues()
+        {
+            m_Transform.position = m_LocalPosition;
+            m_Transform.localScale = m_LocalScale;
+            m_Transform.rotation = m_LocalRotation;
+        }
+
+        /// <summary>
+        /// Sets the currently authorized position, scale, and rotation
+        /// </summary>
+        private void SetTransformValues()
+        {
+            m_LocalPosition = m_Transform.position;
+            m_LocalScale = m_Transform.localScale;
+            m_LocalRotation = m_Transform.rotation;
+        }
 
         /// <inheritdoc/>
         public override void OnNetworkSpawn()
         {
+            m_CachedIsServer = IsServer;
+            m_CachedNetworkManager = NetworkManager;
+
+            CanCommitToTransform = IsServerAuthoritative() ? IsServer : IsOwner;
```

Comment:
This should be added to OnGainedOwnership and OnLostOwnership to support changing ownership after spawning

Reply:
Which I need to add a test for that as well (seeing as the current tests all passed).